soc/silabs/silabs_siwx91x/Kconfig.soc

```
@@ -5,3 +5,30 @@ config SOC_FAMILY_SILABS_SIWX91X
 	bool
 
 rsource "*/Kconfig.soc"
+
+config SIGNING_ENABLED
+  bool "Sign binary with Silicon Labs Commander"
+  default n
+  help
+    Enable signing the Zephyr binary with Silicon Labs Commander during build.
```

Use tabs for indentation, `default n` can be omitted, `SIGNING_ENABLED` is a too generic name as it's specific for a HAL/SoC. Does it even belong in an SoC configuration, how about in `modules/hal_silabs/Kconfig`?

Additionally you'll need to `select BUILD_OUTPUT_BIN`.

```suggestion
config SILABS_COMMANDER_SIGN
	bool "Sign binary with Silicon Labs Commander"
	select BUILD_OUTPUT_BIN
	help
	  Enable signing the Zephyr binary with Silicon Labs Commander during build.
```

---

Only for SoC 91x

---

Resolved this comment